src/@next/TextInput/TextInput.tsx

```diff
@@ -32,6 +32,10 @@ export const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(
       onChange?.(e);
     };
 
+    useEffect(() => {
```

Comment:
The suffix from prop only get accepted once at line 19 as a default value from useState. After that, any change from the suffix won't affect the actual rendered suffix because `suffix` prop that is passed to `<Input />` component is `suffixValue` not the one from the props.

Reply:
I added useEffect to handle the change of suffix prop that get passed to <TextInput />